From 5ac3b99d066e0559cb8777204f8a8eb9f67fec7a Mon Sep 17 00:00:00 2001
Message-Id: <5ac3b99d066e0559cb8777204f8a8eb9f67fec7a.1398812351.git.mark.gross@intel.com>
In-Reply-To: <e87e3533536ae5eec0e06ceb7ef7428e42855790.1398812350.git.mark.gross@intel.com>
References: <e87e3533536ae5eec0e06ceb7ef7428e42855790.1398812350.git.mark.gross@intel.com>
From: Yao Jin <yao.jin@intel.com>
Date: Wed, 26 Feb 2014 22:04:33 +0800
Subject: [PATCH 205/429] Re-Add BayTrail Soc GPIO and LPSS ACPI IDs.

The ACPI ID of gpio driver was removed because there was a bug triggered when gpio driver registered handlers for the ACPI signalled gpio interrupts (caused graphics to crash).
The bug is reported here:
https://bugzilla.kernel.org/show_bug.cgi?id=68291
This patch is just a workaround to add the ACPI IDs back to let the gpio work on BayTrail.

Change-Id: Id27427eb6f00a664db4de4ba1d2a71df284f7b93
---
 drivers/acpi/acpi_lpss.c           | 1 +
 drivers/pinctrl/pinctrl-baytrail.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/acpi/acpi_lpss.c b/drivers/acpi/acpi_lpss.c
index 6745fe1..e603905 100644
--- a/drivers/acpi/acpi_lpss.c
+++ b/drivers/acpi/acpi_lpss.c
@@ -162,6 +162,7 @@ static const struct acpi_device_id acpi_lpss_device_ids[] = {
 	{ "80860F14", (unsigned long)&byt_sdio_dev_desc },
 	{ "80860F41", (unsigned long)&byt_i2c_dev_desc },
 	{ "INT33B2", },
+	{ "INT33FC", },
 
 	{ "INT3430", (unsigned long)&lpt_dev_desc },
 	{ "INT3431", (unsigned long)&lpt_dev_desc },
diff --git a/drivers/pinctrl/pinctrl-baytrail.c b/drivers/pinctrl/pinctrl-baytrail.c
index 665b96b..e599834 100644
--- a/drivers/pinctrl/pinctrl-baytrail.c
+++ b/drivers/pinctrl/pinctrl-baytrail.c
@@ -536,6 +536,7 @@ static const struct dev_pm_ops byt_gpio_pm_ops = {
 
 static const struct acpi_device_id byt_gpio_acpi_match[] = {
 	{ "INT33B2", 0 },
+	{ "INT33FC", 0 },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, byt_gpio_acpi_match);
-- 
1.8.3.2

